### DISCUSSION TITLE
Suggestions, ideas, and feature requests

### DISCUSSION THREAD

**Discussion by daniellovell** [daniellovell/deepmine#2] [Ideas]
Please post any suggestions, ideas, or feature requests here!

**Comment by Berkajerk** [comment]
Horizontal automatic machine driller for boring tunnels. 

-Could be adapted from the automatic vertical miner from the base game.
-Tunnels could eventually have tracks and/or robots to transfer resources from the surface through the tunnels

**Reply by daniellovell** [reply]
Would the in-game conveyors meet the 2nd point? e.g. if you had the tunnel, you could place frames along the floor of the tunnel then use conveyors to transport it back to the surface?

**Reply by Berkajerk** [reply]
Ya something like that. Just throwing out ideas haha.

The game already has the surface mining robot. Could be repurposed to follow a track and pull carts.

**Reply by daniellovell** [reply]
Love it, great ideas. Thank you 😄

**Comment by Saraquael** [comment]
I've seen in the patches.cs that you have hardcoded MinDropQuantity/Y Height value, but since now we can configure max BedrockDepth it would be fantastic if those 5 values adapted to that, maybe calculating them as a percentage of max BedrockDepth.

Alternatively, some kind of logaritmic curve to plot yelds/depth. You known, to stay true to the over-engineering aspect of this game 😀

**Comment by ViroMad** [comment]
So the mod changes to MineCompletionTime and MineAmount; change all drills equally and make them all the same. So the heavy drill which is supposed to be 2x faster is the same speed and mines the same amount as the basic drill. This is probably not a desired result.

I suggest using the configs as multiplier against the drill specs. So the higher tier drills can receive better(or worse stats should they desire to punish themselves) stats.

edit: So I liked the idea so much I cloned the git and did it. I logged the output to bepinex, used a multiplier of 2, and got
```
[Info   :Deep Mine Mod] Drill Name: ItemMiningDrill
[Info   :Deep Mine Mod] Drill Time Was: 0.12 ~ Now is: 0.06
[Info   :Deep Mine Mod] Drill Ammount Was: 0.2 ~ Now is: 0.4
[Info   :Deep Mine Mod] Drill Name: ItemMiningDrillHeavy
[Info   :Deep Mine Mod] Drill Time Was: 0.06 ~ Now is: 0.03
[Info   :Deep Mine Mod] Drill Ammount Was: 0.4 ~ Now is: 0.8
```
